frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -71,7 +71,7 @@ def correct_state
     def ensure_valid_state
       if @order.state != correct_state && !skip_state_validation?
         flash.keep
-        @order.state = correct_state
+        @order.update_column(:state, correct_state)
```

Comment:
@vishalzambre When we do not update state of order here, correct checkout page is rendered but order is in invalid state. For reference: issue #6724